include/aws/common/task_scheduler.h

```diff
@@ -0,0 +1,101 @@
+#ifndef AWS_TASK_SCHEDULER_H
+#define AWS_TASK_SCHEDULER_H
+
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/common/common.h>
+#include <aws/common/priority_queue.h>
+#include <stdint.h>
+
+typedef void *aws_task_arg;
```

Comment:
Does this really need a typedef?

Reply:
Not really. I was wondering myself when porting the code. @JonathanHenson thoughts on this?

Reply:
I'd typedef the function, but not the arg.